packages/lesswrong/components/notifications/NotificationsItem.jsx

```diff
@@ -104,7 +106,16 @@ class NotificationsItem extends Component {
             [classes.unread]: !(notification.createdAt < lastNotificationsCheck || this.state.clicked)
           }
         )}
-        onClick={() => this.setState({clicked: true})}
+        onClick={() => {
+          this.setState({clicked: true})
+          // we also check whether it's a relative link, and if so, scroll to the item
+          const url = new UrlClass(notification.link)
+          const hash = url.hash
+          if (hash) {
```

Comment:
Slightly hacky in the case where this has a hash, but also has a different base URL. Should be fine though.